cylc/flow/loggingutil.py

```diff
@@ -58,17 +58,29 @@ class CylcLogFormatter(logging.Formatter):
     #       deamonise script (url, pid) are not wrapped
     MAX_WIDTH = 999
 
-    def __init__(self, timestamp=True, color=False, max_width=None):
+    def __init__(
+        self, timestamp=True, color=False, max_width=None, dev_info=False
+    ):
         self.timestamp = None
         self.color = None
         self.max_width = self.MAX_WIDTH
         self.wrapper = None
         self.configure(timestamp, color, max_width)
-        # You may find adding %(filename)s %(lineno)d are useful when debugging
-        logging.Formatter.__init__(
-            self,
-            '%(asctime)s %(levelname)-2s - %(message)s',
-            '%Y-%m-%dT%H:%M:%S%Z')
+        if dev_info is True:
+            # Add location message created in codebase when in very very, very
+            # verbose mode.
+            logging.Formatter.__init__(
+                self,
+                (
+                    '[%(pathname)s:%(lineno)d] '
+                    '%(asctime)s %(levelname)-2s - %(message)s'
```

Comment:
Could we have the timestamp first (always the same length).

Reply:
Also, worth checking that this doesn't conflict with the no timestamp mode.

Reply:
Having the timestamp first fixed the conflict which _did_ exist with the `--no-timestamp` option.